packages/admob/ios/RNFBAdmob/RNFBAdMobConsentModule.m

```diff
@@ -0,0 +1,155 @@
+//
+/**
+ * Copyright (c) 2016-present Invertase Limited & Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this library except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#import <React/RCTUtils.h>
+
+#import "RNFBAdMobConsentModule.h"
+#import <RNFBApp/RNFBSharedUtils.h>
+
+@implementation RNFBAdMobConsentModule
+#pragma mark -
+#pragma mark Module Setup
+
+RCT_EXPORT_MODULE();
+
+- (dispatch_queue_t)methodQueue {
+  return dispatch_get_main_queue();
+}
+
+#pragma mark -
+#pragma mark Firebase AdMob Methods
+
+RCT_EXPORT_METHOD(requestInfoUpdate:
+  (NSArray *) publisherIds
+    :(RCTPromiseResolveBlock) resolve
+    :(RCTPromiseRejectBlock) reject
+) {
+  PACConsentInformation *consentInformation = [PACConsentInformation sharedInstance];
+
+  id completionHandler = ^(NSError *error) {
+    if (error != nil) {
+      // todo error
+    } else {
+      resolve(@{
+          @"status": @(consentInformation.consentStatus),
+          @"isRequestLocationInEeaOrUnknown": @(consentInformation.requestLocationInEEAOrUnknown),
+      });
+    }
+  };
+
+  [consentInformation requestConsentInfoUpdateForPublisherIdentifiers:publisherIds completionHandler:completionHandler];
+}
+
+RCT_EXPORT_METHOD(showForm:
+  (NSDictionary *) options
+    :(RCTPromiseResolveBlock) resolve
+    :(RCTPromiseRejectBlock) reject
+) {
+  NSString *privacyPolicy = options[@"privacyPolicy"];
+  PACConsentForm *form = [[PACConsentForm alloc] initWithApplicationPrivacyPolicyURL:[NSURL URLWithString:privacyPolicy]];
+
+  if (options[@"withPersonalizedAds"]) {
+    form.shouldOfferPersonalizedAds = YES;
+  }
+
+  if (options[@"withNonPersonalizedAds"]) {
+    form.shouldOfferNonPersonalizedAds = YES;
+  }
+
+  if (options[@"withAdFree"]) {
+    form.shouldOfferAdFree = YES;
+  }
+
+  id dismissCompletionBlock = ^(NSError *error, BOOL userPrefersAdFree) {
+    if (error != nil) {
+      // todo error
+    } else {
+      resolve(@{
+          @"status": @(PACConsentInformation.sharedInstance.consentStatus),
+          @"userPrefersAdFree": @(userPrefersAdFree),
+      });
+    }
+  };
+
+  [form loadWithCompletionHandler:^(NSError *error) {
+    if (error != nil) {
+      // todo error
+    } else {
+      // todo whats the view controller
+      [form presentFromViewController:self dismissCompletion:dismissCompletionBlock];
```

Comment:
@Salakar how do we get the view controller here? Also does this need to run on the UI thread too?